be/src/storage/rowset/beta_rowset.cpp

```diff
@@ -243,6 +243,7 @@ Status BetaRowset::get_segment_iterators(const vectorized::Schema& schema, const
     seg_options.predicates_for_zone_map = options.predicates_for_zone_map;
     seg_options.use_page_cache = options.use_page_cache;
     seg_options.profile = options.profile;
+    seg_options.runtime_state = options.runtime_state;
```

Comment:
Why put the state into an option?

Reply:
> Why put the state into an option?

we didn't pass `RuntimeState` in `ChunkIterator::get_next`, so I add it into `SegmentReadOptions` so that segment iterator can finish in time if query is cancelled when reading